<?xml version="1.0"?>
<launch>

    <arg name="dev" default="/dev/roboclaw"/>
    <arg name="baud" default="460800"/>
    <arg name="address" default="128"/>
    <arg name="max_abs_linear_speed" default="1.5"/>
    <arg name="max_abs_angular_speed" default="1.5"/>
<!-- Wheels max acceleration m/s^2 (TODO: split in angular and linear acceleration) -->
    <arg name="acc_lim" default="1.0"/>
<!-- 1957 ticks/rev ; 0.2 cm diam ; 0.6283186 m/rev -->
    <arg name="ticks_per_meter" default="3114.66"/>
    <arg name="base_width" default="0.4"/>
    <arg name="run_diag" default="true"/>

    <node if="$(arg run_diag)" pkg="roboclaw_node" type="roboclaw_sim.py" name="roboclaw_node">
        <param name="~dev" value="$(arg dev)"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~address" value="$(arg address)"/>
        <param name="~max_abs_linear_speed" value="$(arg max_abs_linear_speed)"/>
	<param name="~max_abs_angular_speed" value="$(arg max_abs_angular_speed)"/>
	<param name="~acc_lim" value="$(arg acc_lim)"/>
        <param name="~ticks_per_meter" value="$(arg ticks_per_meter)"/>
        <param name="~base_width" value="$(arg base_width)"/>
    </node>


</launch>
